tests/publiccloud/check_registercloudguest.pm

```diff
@@ -19,7 +19,9 @@ use strict;
 use utils;
 use publiccloud::utils;
 use publiccloud::ssh_interactive 'select_host_console';
-use version_utils 'is_sle';
+
+my $path = is_sle('>15') && is_sle('<15-SP3') ? '/usr/sbin' : '';
```

Comment:
I just realize that unfortunately your "best practice"  regarding ending slash won't fly :man_shrugging:  . As always there was good reason why previously this string was `/usr/sbin/` . The problem comes when condition is false and path become '' . Previously this would  make line 54 below execute `sudo registercloudguest --clean` which is totally valid command and makes bash to look through $PATH variable and find `registercloudguest` executable in right place and execute it but with your change now we have different story `sudo /registercloudguest --clean` :man_shrugging:

Reply:
Or you know, we can change the default in the ternary operator to `/bin' or something else (instead of empty string) that makes sense instead of relying on the way BASH replaces variables in an obfuscated manner.

Also, what are the odds that this test does NOT run on SLE? My understanding was that this is SLE-specific.

Reply:
> Or you know, we can change the default in the ternary operator to `/bin' or something else (instead of empty string) that makes sense instead of relying on the way BASH replaces variables in an obfuscated manner.

this make sense when your goal is just to make wheels rollings in most stable and solid way . But it is NOT  our goal instead we trying to prove that some code works as expected. And one of many things this **test** is proving that `registercloudguest` binary is available in $PATH . By hardcoding absolute path we will lower test coverage and if execution of `registercloudguest` w/o absolute path  will get broken we will not notice this :shrug: 

> Also, what are the odds that this test does NOT run on SLE? My understanding was that this is SLE-specific.

` is_sle('>15') && is_sle('<15-SP3')`  - means  SLE version should not be lower then 15 SP0 and SLE version should not be higher then SLE 15 SP3 which gives you SLE 15 SP0 , SLE 15 SP1 and SLE 15 SP2